### PR TITLE
add openvpn-client

### PR DESCRIPTION
**Pull request by peterthomassen**
Last thing to fix that I am aware of is a potential IP subnet collision on stack and slave. That's not currently an issue, but it could become one if the slave has 7 containers (currently 4).

### CODE REVIEW

File: openvpn-client/conf/client.conf [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+# Are we connecting to a TCP or
+# UDP server?  Use the same setting as
+# on the server.
+;proto tcp
+proto udp
+
+# The hostname/IP and port of the server.
+# You can have multiple remote entries
+# to load balance between the servers.
+remote desec.sandbox.dedyn.io 1194
```
Comment (nils-wisiol) [inline openvpn-client/conf/client.conf]: needs updating

File: openvpn-client/conf/client.conf [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+# to the internet such as laptops.
+resolv-retry infinite
+
+# Most clients don't need to bind to
+# a specific local port number.
+nobind
+
+# Downgrade privileges after initialization (non-Windows only)
+;user nobody
+;group nogroup
```
Comment (nils-wisiol) [inline openvpn-client/conf/client.conf]: we should do this if possible

**Review by nils-wisiol — COMMENTED** [review]
I'm voting to merge desec-slave into desec-stack, for the following reasons:

- currently, there is quite some code duplication among this PR and desec-stack/#282
- these two PRs need to merged and deployed together anyhow
- subnet configuration will be less error-prone

File: openvpn-client/conf/client.conf (hunk 2 above)
Comment (peterthomassen) [reply openvpn-client/conf/client.conf]: good catch, done

File: openvpn-client/conf/client.conf (hunk 1 above)
Comment (peterthomassen) [reply openvpn-client/conf/client.conf]: done. It's an environment variable now

**Comment by peterthomassen** [comment]
I am slightly inclined against merging this repository into the stack: For example, once we merge desec-io/desec-stack#282, the stack will be capable of both MySQL replication and VPN access to nsmaster. We can then keep the old slaves running with MySQL replication, and already work on switching the desec-slave repository to the new replication mechanism where MySQL support is entirely removed (and instead AXFR and a key-value backend is used).

If we merge this repo into the stack, than we will have to coordinate more larger pieces in a single PRs, because stack and slave are required to be more in sync, which is not the case now. (As I said, the MySQL replication has undergone no change so far, so existing slaves would continue to work.)

These are my 2cts which I wanted to write down for you to consider; otherwise, no stakes involved, and I'll go with whatever you think is best.

**Comment by nils-wisiol** [comment]
This PR should add some docs explaining how to integrate the slave into an existing stack infrastructure.

**Comment by peterthomassen** [comment]
done!

**Comment by peterthomassen** [comment]
incorporating in #19

**Pull request closed without merging**